acceptance/tests/test_upgrade_puppet5_to_puppet6.rb

```diff
@@ -48,7 +48,7 @@ class { puppet_agent:
     agents_only.each do |agent|
       on(agent, puppet('agent -t --debug'), acceptable_exit_codes: 2)
       wait_for_installation_pid(agent)
-      assert_agent_version_on(agent, latest_version.scan(%r{6\.\d*\.\d*\.\d*}).first)
+      assert(puppet_agent_version_on(agent) =~ %r{^6\.\d+.\d+.*$})
```

Comment:
Nit we lost an escape on the dot, also the `.*$` is a little redundant since we don't care what comes after:

```suggestion
      assert(puppet_agent_version_on(agent) =~ %r{^6\.\d+\.\d+})
```